types/directive.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+// Directive is a representation of the GraphQL Directive.
+//
+// http://spec.graphql.org/draft/#sec-Language.Directives
+type Directive struct {
+	Name      Ident
+	Arguments ArgumentList
+}
+
+// DirectiveDefinition is a representation of the GraphQL DirectiveDefinition.
+//
+// http://spec.graphql.org/draft/#sec-Type-System.Directives
+type DirectiveDefinition struct {
+	Name      string
```

Comment:
we should probably be consistent with our usage of `Ident`. Some types have `Ident` while others are `string`s. All type definition names should use the `Ident` type so that its location can be exposed.

Reply:
Can you give an example, please?